chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc

```diff
@@ -91,6 +92,52 @@ void RenderViewContextMenu::RegisterMenuShownCallbackForTesting(
 #undef RenderViewContextMenu
 #undef BRAVE_APPEND_SEARCH_PROVIDER
 
+namespace {
+
+#if BUILDFLAG(ENABLE_TOR)
+bool HasAlreadyOpenedTorWindow(Profile* profile) {
+  for (Browser* browser : *BrowserList::GetInstance()) {
+    if (browser->profile()->IsTor() &&
+        browser->profile()->GetOriginalProfile() == profile)
+      return true;
+  }
+
+  return false;
+}
+
+// Modified OnProfileCreated() in render_view_context_menu.cc
+// to handle additional |use_new_tab| param.
+void OnTorProfileCreated(const GURL& link_url,
+                         const content::Referrer& referrer,
+                         bool use_new_tab,
+                         Profile* profile,
+                         Profile::CreateStatus status) {
+  if (status == Profile::CREATE_STATUS_INITIALIZED) {
+    Browser* browser = chrome::FindLastActiveWithProfile(profile);
+    /* |ui::PAGE_TRANSITION_TYPED| is used rather than
+       |ui::PAGE_TRANSITION_LINK| since this ultimately opens the link in
+       another browser. This parameter is used within the tab strip model of
+       the browser it opens in implying a link from the active tab in the
+       destination browser which is not correct. */
+    NavigateParams nav_params(browser, link_url, ui::PAGE_TRANSITION_TYPED);
+    if (use_new_tab) {
+      nav_params.disposition = WindowOpenDisposition::NEW_FOREGROUND_TAB;
+    } else {
+      // Stop NTP loading to show tab throbber wait spinning till tor is
+      // initialized.
+      browser->tab_strip_model()->GetActiveWebContents()->Stop();
```

Comment:
kinda sketchy. can we check if it's a NTP WebContents? maybe also check for `GetActiveWebContents != nullptr` to not crash if something goes the other way?